### PR TITLE
Solve localhost issues on default server client discovery setup via environmental variable

### PR DESCRIPTION
**Pull request by MiguelBarro**
This PR is related with: https://github.com/eProsima/Fast-RTPS/pull/1160

The **ROS2_AUTO_CLIENT_SERVER** or **FASTDDS_AUTO_CLIENT_SERVER** environment variables are used to specify the server address. Now:
+ the server may use the addresses 127.0.0.1 or 0.0.0.0 to listening in all local machine interfaces.
+ the client may use the addresses 127.0.0.1 or 0.0.0.0 to be indicated the server is local.

To summarize one can setup intermachine communication by first launching a server as 127.0.0.1 or 0.0.0.0.
If a client launch later:
+ shares machine with the server it suffices to use 127.0.0.1 or 0.0.0.0 to communicate.
+ is in another machine must specify the real IP address of any of the server machine interfaces.
+ shares machine with the

### CODE REVIEW

**Comment by MiguelCompany** [comment]
> Everything works fine and in single machine, and across machines. As discussed, it'd be nice to have a unit test for `RTPSParticipantImpl::did_mutation_took_place_on_meta()`

We'll leave that unit test for a separate PR

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2014/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1752/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2008/)
